app/validators/questionnaire_schema.py

```diff
@@ -0,0 +1,298 @@
+import collections
+from collections import defaultdict
+from functools import cached_property, lru_cache
+import jsonpath_rw_ext as jp
+from jsonpath_rw import parse
+
+
+def has_default_route(routing_rules):
+    for rule in routing_rules:
+        if "goto" not in rule or "when" not in rule["goto"].keys():
+            return True
+    return False
+
+
+def get_routing_when_list(routing_rules):
+    when_list = []
+    for rule in routing_rules:
+        when_clause = rule.get("goto", {})
+        when_list.append(when_clause)
+    return when_list
+
+
+def is_contained_in_list(dict_list, key_id):
```

Comment:
I think `in_list` is misleading
```suggestion
def is_contained_in_dict_list(dict_list, key_id):
```